### ISSUE TITLE
Setting the appResourcesPath or nsconfig through command line

### ISSUE THREAD

**Issue by farfromrefug** [NativeScript/nativescript-cli#4718]
Is there a way to do this right now?
I want to build a "wearos" variant of my app. But i need a different App_Resources folder and different app id.
I am trying to achive this through package.json scripts. But to do that i need command line args to switch the folders

**Comment by rosen-vladimirov** [comment]
Hey @farfromrefug ,
Currently the id of the application is set in the package.json file, while the App_Resources can be set in nsconfig.json, as described [in the docs](https://docs.nativescript.org/core-concepts/project-structure#the-nsconfigjson-file).
So for your case, you can use script to change both values and build the app after that. Does it sound reasonable for your case?

**Comment by farfromrefug** [comment]
@rosen-vladimirov someone advised that but I dont like that that solution. the thing Is that it modifies committed files. I get it is not in your timeline but it would be a great feature to have

**Comment by rosen-vladimirov** [comment]
So, just to clear this out, you would like to pass some `--` flag to CLI's command, which will change the nsconfig.json which will be used for building the application.
Also, the id should be persisted in the nsconfig.json instead of package.json.

Am I right?

**Comment by farfromrefug** [comment]
yes the id in the nsconfig would be great. and the cli  arc that I would like is one to a specify the nsconfig  file to use. just like tsc  and the tsconfig file. thanks

**Comment by Eonfuzz** [comment]
Any updates on this? It's 🌶 and would our dev pipeline cleaner

**Comment by fbeuKiel** [comment]
That would be great...and should solve [my problem](https://stackoverflow.com/questions/59682642/how-to-manipulate-the-projectidentifiers-app-identifier-in-the-before-prepare) to dynamically set an application id (e.g. with a symbolic link)

**Comment by pfumagalli** [comment]
I do this with an environment variable.

the `id` and `appResourcesPath` are configurable via `nativescript.config.ts` and when this is executed, the code determines the correct values for those parameters.

I have two _profiles_ of my app (all different resources, configurations, ...) and my tree looks like this:

```
+- resources (instead of App_Resources)
   +- flavor1
   |  +- Android
   |  +- iOS
   +- flavour2
      +- Android
      +- iOS
```

I use the `APP_FLAVOUR` environment variable to determine which flavour to build and my `nativescript.config.ts` looks like this:

```ts
import { NativeScriptConfig } from '@nativescript/core'

/* Get (and ensure we have the default) the "APP_FLAVOUR" environment variable */
const flavour = process.env.APP_FLAVOUR = process.env.APP_FLAVOUR || 'flavour1'
switch (flavour) {
  case 'flavour1': break
  case 'flavour2': break
  default:
    throw new Error(`Unknown / unsupported app flavor "${flavour}" (must be "flavour1" or "flavour2")`)
}

/* Export our NativeScript configuration */
export default {
  id: `foo.bar.${flavor}.myapp`,
  appResourcesPath: `resources/${flavor}`,
  // ... all the rest of the stuff
} as NativeScriptConfig
```

I also modify `webpack.config.js` to push my _flavour_ as a constant in the app

```js
new webpack.DefinePlugin({
  // ... all various definitions...
  'TNS_ENV': JSON.stringify(mode),
  'APP_FLAVOUR': JSON.stringify(process.env.APP_FLAVOUR),
})
```

And finally my `types/env.d.ts` file looks like this:

```ts
declare const TNS_ENV: string
declare const APP_FLAVOUR: 'flavour1' | 'flavour2'
```

So that anywhere in the code I can simply refer to the `APP_FLAVOUR` constant and handle some special cases...
